Python/wolfssl-python-3.8.5.patch

```diff
@@ -0,0 +1,1995 @@
++                    with self.assertRaisesRegex(
++                           ssl.CertificateError,
++                            "peer subject name mismatch"):
++                        s.connect((HOST, server.port))
++                else:
++                    with self.assertRaisesRegex(
++                            ssl.CertificateError,
++                            "Hostname mismatch, certificate is not valid for 'invalid'."):
++                        s.connect((HOST, server.port))
+ 
+         # missing server_hostname arg should cause an exception, too
+         server = ThreadedEchoServer(context=server_context, chatty=True)
+@@ -2964,7 +3152,9 @@
+     def test_ecc_cert(self):
+         client_context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
+         client_context.load_verify_locations(SIGNING_CA)
+-        client_context.set_ciphers('ECDHE:ECDSA:!NULL:!aRSA')
++        if not ssl.IS_WOLFSSL:
++            client_context.set_ciphers('ECDHE:ECDSA:!NULL:!aRSA')
++        
+         hostname = SIGNED_CERTFILE_ECC_HOSTNAME
+ 
+         server_context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
+@@ -2989,13 +3179,19 @@
+         #       algorithms.
+         client_context.options |= ssl.OP_NO_TLSv1_3
+         # only ECDSA certs
+-        client_context.set_ciphers('ECDHE:ECDSA:!NULL:!aRSA')
++        if ssl.IS_WOLFSSL:
++            # wolfSSL doesn't support cipher rule
++            client_context.set_ciphers('ECDHE-ECDSA-AES256-GCM-SHA384')
++        else:
++            client_context.set_ciphers('ECDHE:ECDSA:!NULL:!aRSA')
+         hostname = SIGNED_CERTFILE_ECC_HOSTNAME
+-
++        
+         server_context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
+         # load ECC and RSA key/cert pairs
+         server_context.load_cert_chain(SIGNED_CERTFILE_ECC)
+-        server_context.load_cert_chain(SIGNED_CERTFILE)
++        if not ssl.IS_WOLFSSL:
++            # wolfSSL handles one certificate
++            server_context.load_cert_chain(SIGNED_CERTFILE)
+ 
+         # correct hostname should verify
+         server = ThreadedEchoServer(context=server_context, chatty=True)
+@@ -3116,21 +3312,38 @@
+         with server, \
+              client_context.wrap_socket(socket.socket(),
+                                         server_hostname=hostname) as s:
+-            # TLS 1.3 perform client cert exchange after handshake
+-            s.connect((HOST, server.port))
+-            try:
+-                s.write(b'data')
+-                s.read(4)
+-            except ssl.SSLError as e:
+-                if support.verbose:
+-                    sys.stdout.write("\nSSLError is %r\n" % e)
+-            except OSError as e:
+-                if e.errno != errno.ECONNRESET:
+-                    raise
+-                if support.verbose:
+-                    sys.stdout.write("\nsocket.error is %r\n" % e)
++            if ssl.IS_WOLFSSL:
++                # wolfSSL do_handshake performs client cert exchaange while doing handshaking
```

Comment:
We do support post-handshake client auth, though, right? We're just not using it here? May be worth clarifying.